packages/components/src/palette-edit/test/index.tsx

```diff
@@ -82,18 +83,282 @@ describe( 'getNameForPosition', () => {
+			name: 'Midnight',
+			slug: 'midnight',
+		},
+	];
+
+	it( 'shows heading label', () => {
+		render( <PaletteEdit { ...defaultProps } colors={ colors } /> );
+
+		const paletteLabel = screen.getByRole( 'heading', {
+			level: 2,
+			name: 'Test label',
+		} );
+
+		expect( paletteLabel ).toBeInTheDocument();
+	} );
+
+	it( 'shows heading label with custom heading level', () => {
+		render(
+			<PaletteEdit
+				{ ...defaultProps }
+				colors={ colors }
+				paletteLabelHeadingLevel={ 5 }
+			/>
+		);
+
+		expect(
+			screen.getByRole( 'heading', {
+				level: 5,
+				name: 'Test label',
+			} )
+		).toBeInTheDocument();
+	} );
+
+	it( 'shows empty message', () => {
+		render(
+			<PaletteEdit
+				{ ...defaultProps }
+				emptyMessage={ 'Test empty message' }
+			/>
+		);
+
+		expect( screen.getByText( 'Test empty message' ) ).toBeInTheDocument();
+	} );
+
+	it( 'can remove all colors', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } colors={ colors } /> );
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Color options',
+			} )
+		);
+
+		expect(
+			screen.getByRole( 'button', {
+				name: 'Remove all colors',
+			} )
+		).toBeInTheDocument();
+	} );
+
+	it( 'can reset colors', async () => {
+		const user = userEvent.setup();
+		render(
+			<PaletteEdit { ...defaultProps } colors={ colors } canReset />
+		);
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Color options',
+			} )
+		);
+		expect(
+			screen.getByRole( 'button', {
+				name: 'Reset colors',
+			} )
+		).toBeInTheDocument();
+	} );
+
+	it( 'can not reset colors', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } colors={ colors } /> );
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Color options',
+			} )
+		);
+		expect(
+			screen.queryByRole( 'button', {
+				name: 'Reset colors',
+			} )
+		).not.toBeInTheDocument();
+	} );
+
+	it( 'can add a new color palette', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } colors={ colors } /> );
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Add color',
+			} )
+		);
+
+		await waitFor( () => {
+			expect( defaultProps.onChange ).toHaveBeenCalledWith( [
+				...colors,
+				{
+					color: '#000',
+					name: 'Color 1',
+					slug: 'color-1',
+				},
+			] );
+		} );
+	} );
+
+	it( 'can add a new graident palette', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } gradients={ gradients } /> );
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Add gradient',
+			} )
+		);
+
+		await waitFor( () => {
+			expect( defaultProps.onChange ).toHaveBeenCalledWith( [
+				...colors,
+				{
+					color: '#000',
+					name: 'Color 1',
+					slug: 'color-1',
+				},
+			] );
+		} );
+	} );
+
+	it( 'can not add a new palette', () => {
```

Comment:
```suggestion
	it( 'can not add new colors when `canOnlyChangeValues` is enabled', () => {
```

Reply:
I believe it's important to check both aspects — either in the same test, or in two separate tests. One checks that our public APIs work as expected, the other one tests that the component works as intended for the end user.

We could technically omit the `onChange` check if the component is tested in controlled mode, because if `onChange` didn't work as expected, the component wouldn't work for the end user either.

But I still prefer to test `onChange` (and the number of times it's been called) because in my experience it's also a great proxy for other undetected issues.

Reply:
Agreed. Not a blocker for this particular PR though.